src/main/java/uk/gov/hmcts/darts/arm/helper/DataStoreToArmHelper.java

```diff
@@ -142,7 +153,10 @@ public ExternalObjectDirectoryEntity createArmExternalObjectDirectoryEntity(Exte
         armExternalObjectDirectoryEntity.setCreatedBy(userAccount);
         armExternalObjectDirectoryEntity.setLastModifiedBy(userAccount);
         armExternalObjectDirectoryEntity.setTransferAttempts(1);
-
+        // This is only set for DETS records
+        if (nonNull(externalObjectDirectory.getOsrUuid())) {
+            armExternalObjectDirectoryEntity.setOsrUuid(externalObjectDirectory.getOsrUuid());
+        }
```

Comment:
since this is meant to be a generic method, can't this DETS specific change be move up a level to the DETS code that is calling this?

Reply:
I added this just to highlight the difference between DETS and Unstructured. Its doesn't actually need the nonNull check.

Reply:
I was not referring to the null check, the whole block is very DETS specific I think doesn't really belong here for that reason

Reply:
No it isn't. This is shared by both unstructured and dets where they copy the original eod and make a new arm one. All I was trying to highlight is that the setting the setOsrUuid should be null for unstructured but have a value for dets.